ssl/statem/statem_srvr.c

```diff
@@ -1735,16 +1735,14 @@ static int tls_early_post_process_client_hello(SSL_CONNECTION *s)
         s->client_version = clienthello->legacy_version;
     }
     /*
-     * Do SSL/TLS version negotiation if applicable. For DTLS we just check
-     * versions are potentially compatible. Version negotiation comes later.
+     * Do SSL/TLS version negotiation if applicable.
      */
-    if (!SSL_CONNECTION_IS_DTLS(s)) {
-        protverr = ssl_choose_server_version(s, clienthello, &dgrd);
-    } else if (ssl->method->version != DTLS_ANY_VERSION &&
-               DTLS_VERSION_LT((int)clienthello->legacy_version, s->version)) {
+    if (SSL_CONNECTION_IS_DTLS(s)
+            && ssl->method->version != DTLS_ANY_VERSION
+            && DTLS_VERSION_LT((int)clienthello->legacy_version, s->version)) {
         protverr = SSL_R_VERSION_TOO_LOW;
```

Comment:
This check seems redundant to me. AFAICT isn't exactly the same check performed by `ssl_choose_server_version` in the first switch? The only difference being that it will end up with SSL_R_WRONG_SSL_VERSION instead of SSL_R_VERSION_TOO_LOW. I'm not sure if this difference is something we care about.

Reply:
Ok sure, I didn't dare to go all the way. I think it is preferable that all version logic is contained in `ssl_choose_server_version`, so i'll take your suggestion. Do we need a nod from @t8m ?

Reply:
I'd say just go ahead and make the change, if you agree with my analysis.